templates/default/deploy-scripts/src/main.rs

```diff
@@ -0,0 +1,162 @@
+pub mod deployer;
+use anyhow::Context;
+use clap::Parser;
+use concordium_rust_sdk::{
+    common::types::Amount,
+    endpoints::{self, RPCError},
+    smart_contracts::{
+        common::{self as contracts_common, NewContractNameError, NewReceiveNameError},
+        types::{OwnedContractName, OwnedParameter, OwnedReceiveName},
+    },
+    types::{
+        smart_contracts::{ExceedsParameterSize, WasmModule},
+        transactions,
+        transactions::send::GivenEnergy,
+    },
+    v2,
+};
+
+use concordium_rust_sdk::types::transactions::InitContractPayload;
+use deployer::{Deployer, ModuleDeployed};
+use hex::FromHexError;
+use std::{
+    io::Cursor,
+    path::{Path, PathBuf},
+};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum DeployError {
+    #[error("concordium error: {0}")]
+    RPCError(#[from] RPCError),
+    #[error("transport error: {0}")]
+    TransportError(#[from] v2::Error),
+    #[error("query error: {0}")]
+    QueryError(#[from] endpoints::QueryError),
+    #[error("anyhow error: {0}")]
+    AnyhowError(#[from] anyhow::Error),
+    #[error("There are unfinalized transactions. Transaction nonce is not reliable enough.")]
+    NonceNotFinal,
+    #[error("Transaction rejected: {0}")]
+    TransactionRejected(RPCError),
+    #[error("Transaction rejected: {0:?}")]
+    TransactionRejectedR(String),
+    #[error("Invalid block item: {0}")]
+    InvalidBlockItem(String),
+    #[error("Invalid contract name: {0}")]
+    InvalidContractName(String),
+    #[error("hex decoding error: {0}")]
+    HexDecodingError(#[from] FromHexError),
+    #[error("failed to parse receive name: {0}")]
+    FailedToParseReceiveName(String),
+    #[error("Json error: {0}")]
+    JSONError(#[from] serde_json::Error),
+    #[error("Parameter size error: {0}")]
+    ParameterSizeError(#[from] ExceedsParameterSize),
+    #[error("Receive name error: {0}")]
+    ReceiveNameError(#[from] NewReceiveNameError),
+    #[error("Contract name error: {0}")]
+    ContractNameError(#[from] NewContractNameError),
+    #[error("Reqwest error: {0}")]
+    ReqwestError(#[from] reqwest::Error),
+    #[error("Invalid metadata hash: {0}")]
+    InvalidHash(String),
+    #[error("IO error: {0}")]
+    IOError(#[from] std::io::Error),
+    #[error("Invoke contract failed: {0}")]
+    InvokeContractFailed(String),
+}
+
+/// Reads the wasm module from a given file path and file name.
+fn get_wasm_module(file: &Path) -> Result<WasmModule, DeployError> {
+    let wasm_module = std::fs::read(file).context("Could not read the WASM file")?;
+    let mut cursor = Cursor::new(wasm_module);
+    let wasm_module: WasmModule = concordium_rust_sdk::common::from_bytes(&mut cursor)?;
+    Ok(wasm_module)
+}
+
+/// Command line flags.
+#[derive(clap::Parser, Debug)]
+#[clap(author, version, about)]
+struct App {
```

Comment:
Perhaps we should add subcommands so that you have to call `cargo run -- deploy ...` or `cargo run -- update` etc?

Reply:
That would make it easier for developers to extend this boilerplate with more functions.

Reply:
I meant the latter, but I see that my examples were ambiguous. 
Ok, you've convinced me that the current approach is sufficient ;)